runtime/commonMain/src/kotlinx/serialization/SealedSerializer.kt

```diff
@@ -112,11 +112,11 @@ internal class SealedClassSerializer<T : Any>(
             }.mapValues { it.value.value }
     }
 
-    override fun findPolymorphicSerializer(decoder: CompositeDecoder, klassName: String): DeserializationStrategy<out T> {
-        return serialName2Serializer[klassName] ?: super.findPolymorphicSerializer(decoder, klassName)
+    override fun findPolymorphicSerializerOrNull(decoder: CompositeDecoder, klassName: String?): DeserializationStrategy<out T>? {
+        return serialName2Serializer[klassName] ?: super.findPolymorphicSerializerOrNull(decoder, klassName)
     }
 
-    override fun findPolymorphicSerializer(encoder: Encoder, value: T): SerializationStrategy<T> {
-        return (class2Serializer[value::class] ?: super.findPolymorphicSerializer(encoder, value)).cast()
+    override fun findPolymorphicSerializerOrNull(encoder: Encoder, value: T): SerializationStrategy<T>? {
```

Comment:
Also, maybe I oversaw something, but why do we have to make this method non-throwing at all? Only to support alternative exception text in JSON? Looks like an overkill.

However, the nullable result is more future-proof, e.g. for formats/serializers that imply some default behavior regardless of the module. Not sure we have to encourage such design.

Reply:
To allow other formats (e.g. it's easy to see XML wants to use it) leveraging it.
We already rely on it, it's still internal, so doesn't seem to be a problem anyway